tests/functional/contact.js

```diff
@@ -0,0 +1,133 @@
+
+    casper.start(communitiesUrl, function() {
+        test.assertHttpStatus(200);
+        test.assertExists('#map', 'Map exists');
+        test.assertElementCount('.category-tabs li a', 2, 'Tab navigation exists');
+        test.assertVisible('#map .legend', 'Communities legend is visible');
+        test.assertExists('#nav-spaces', 'Spaces navigation exists');
+        test.assertNotVisible('#nav-spaces', 'Spaces navigation is not visible');
+        test.assertExists('#nav-communities', 'Communities navigation exists');
+        test.assertVisible('#nav-communities', 'Communities navigation is visible');
+        test.assertNotVisible('#nav-spaces-select', 'Mobile navigation is not visible');
+        test.assertExists('.category-tabs > li[data-id="communities"].current', 'Communities tab is selected');
+    });
+
+    casper.run(function() {
+        test.done();
+        helpers.done();
+    });
+});
+
+casper.test.begin('Contact, Spaces navigation click: ' + spacesUrl, 2, function suite(test) {
+
+    casper.options.waitTimeout = 10000;
+
+    casper.start(spacesUrl, function() {
+        this.click('#nav-spaces li[data-id="mountain-view"] > a');
+    });
+
+    casper.waitForUrl(/mountain-view/, function() {
+        test.assertUrlMatch(/mountain-view/, 'URL changed sucessfully');
+    });
+
+    casper.waitForSelector('#entry-container #mountain-view', function() {
+        test.assert(true, 'Page content updated successfully');
+    });
+
+    casper.run(function() {
+        test.done();
+        helpers.done();
+    });
+});
+
+casper.test.begin('Contact, Tab navigation click: ' + spacesUrl, 3, function suite(test) {
+
+    casper.options.waitTimeout = 10000;
+
+    casper.start(spacesUrl, function() {
+        this.click('.category-tabs li[data-id="communities"] > a');
+    });
+
+    casper.waitForUrl(/communities/, function() {
+        test.assertUrlMatch(/communities/, 'URL changed sucessfully');
+        test.assertVisible('#map .legend', 'Communities legend is visible');
+    });
+
+    casper.waitForSelector('#entry-container #communities', function() {
+        test.assert(true, 'Page content updated successfully');
+    });
+
+    casper.run(function() {
+        test.done();
+        helpers.done();
+    });
+});
+
+casper.test.begin('Contact, Mobile interaction: ' + spacesUrl, 4, function suite(test) {
+
+    casper.options.viewportSize = config.viewport.mobile;
+    casper.options.waitTimeout = 10000;
+
+    casper.start(url, function() {
+        test.assertNotVisible('#map', 'Map is not visible on mobile');
+        test.assertVisible('#nav-spaces-select', 'Mobile navigation is visible');
+    });
+
+    casper.thenEvaluate(function(space) {
+        var select = document.getElementById('nav-spaces-select');
+        select.value = space;
+
+        var evt = document.createEvent('Event');
+        evt.initEvent('change', true, true);
+        select.dispatchEvent(evt);
```

Comment:
We might need to do this kinda thing i.e. dispatch events, in more places as we move forward so, perhaps it is worth wrapping the above three lines in a triggerEvent function inside helpers.js ?